src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -198,259 +198,171 @@ public RestoreCommand(IApplicationEnvironment env)
+            context.ProjectLibraryProviders.Add(new LocalDependencyProvider(
+                new PackageSpecReferenceDependencyProvider(packageSpecResolver)));
 
-            var effectiveSources = PackageSourceUtils.GetEffectivePackageSources(
-                SourceProvider,
-                FeedOptions.Sources,
-                FeedOptions.FallbackSources);
+            context.LocalLibraryProviders.Add(
+                new LocalDependencyProvider(
+                    new NuGetDependencyResolver(nugetRepository)));
 
-            AddRemoteProvidersFromSources(remoteProviders, effectiveSources);
+            var effectiveSources = PackageSourceUtils.GetEffectivePackageSources(SourceProvider,
+                FeedOptions.Sources, FeedOptions.FallbackSources);
 
-            var tasks = new List<Task<GraphNode>>();
+            AddRemoteProvidersFromSources(context.RemoteLibraryProviders, effectiveSources);
 
-            if (useLockFile)
-            {
-                Reports.Information.WriteLine(string.Format("Following lock file {0}", projectLockFilePath.White().Bold()));
+            var remoteWalker = new RemoteDependencyWalker(context);
 
-                var context = new RestoreContext
-                {
-                    FrameworkName = ApplicationEnvironment.RuntimeFramework,
-                    ProjectLibraryProviders = projectProviders,
-                    LocalLibraryProviders = localProviders,
-                    RemoteLibraryProviders = remoteProviders,
-                };
+            var tasks = new List<Task<GraphNode<RemoteResolveResult>>>();
 
-                contexts.Add(context);
+            if (useLockFile)
+            {
+                Logger.WriteInformation(string.Format("Following lock file {0}", projectLockFilePath.White().Bold()));
 
                 foreach (var lockFileLibrary in lockFile.Libraries)
                 {
                     var projectLibrary = new LibraryRange
                     {
                         Name = lockFileLibrary.Name,
-                        VersionRange = new SemanticVersionRange
-                        {
-                            MinVersion = lockFileLibrary.Version,
-                            MaxVersion = lockFileLibrary.Version,
-                            IsMaxInclusive = true,
-                            VersionFloatBehavior = SemanticVersionFloatBehavior.None,
-                        }
+                        VersionRange = new VersionRange(minVersion: lockFileLibrary.Version,
+                            floatRange: new FloatRange(NuGetVersionFloatBehavior.None))
                     };
-                    tasks.Add(restoreOperations.CreateGraphNode(context, projectLibrary, _ => false));
+                    // DNU REFACTORING TODO: replace the hardcoded framework name with IRuntimeEnvironment.RuntimeFramework
+                    tasks.Add(remoteWalker.Walk(projectLibrary, new NuGetFramework("DNX,Version=v4.5.1")));
```

Comment:
There should be a constant in `NuGet.Frameworks.FrameworkConstants` for DNX 4.5.1